### PR TITLE
Some refactoring/renames for binding extensibility

### PR DESCRIPTION
**Pull request by mathewc**

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ namespace Microsoft.Azure.WebJobs.Host.Bindings
 {
     // General rule for binding parameters to an AsyncCollector. 
     // Supports the various flavors like IAsyncCollector, ICollector, out T, out T[]. 
-    internal class AsyncCollectorBindingProvider<TAttribute, TType> : FluentBindingProvider<TAttribute>, IBindingProvider, IRuleProvider
+    internal class AsyncCollectorBindingProvider<TAttribute, TType> : FluentBindingProvider<TAttribute>, IBindingProvider, IBindingRuleProvider
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/AsyncCollectorBindingProvider.cs]: I think IRule/IRuleProvider were too general

File: test/Microsoft.Azure.WebJobs.Host.UnitTests/JobHostQueuesConfigurationTests.cs [hunk 2]
```diff
@@ -50,5 +52,25 @@ public void VisibilityTimeout_CanGetAndSetValue()
             config.VisibilityTimeout = TimeSpan.FromSeconds(30);
             Assert.Equal(TimeSpan.FromSeconds(30), config.VisibilityTimeout);
         }
+
+        [Fact]
+        public void JsonSerialization()
+        {
+            var jo = new JObject
+            {
+                { "MaxPollingInterval", 5000 }
```
Comment (mathewc) [inline test/Microsoft.Azure.WebJobs.Host.UnitTests/JobHostQueuesConfigurationTests.cs]: Test verifying that we can deserialize both formats

File: src/Microsoft.Azure.WebJobs.Host/JobHostQueuesConfiguration.cs [hunk 3]
```diff
@@ -89,7 +89,7 @@ public int NewBatchThreshold
         /// Gets or sets the longest period of time to wait before checking for a message to arrive when a queue remains
         /// empty.
         /// </summary>
-        [JsonIgnore]
+        [JsonConverter(typeof(MillisecondTimeSpanConverter))]
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs.Host/JobHostQueuesConfiguration.cs]: >MillisecondTimeSpanConverter [](start = 30, length = 28)

nice :)

**Comment by MikeStall** [comment]
:shipit:

**Comment by mathewc** [comment]
Merged

**Pull request merged**